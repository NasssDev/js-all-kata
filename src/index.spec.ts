import runGoldenMaster from "jest-golden-master";
import { Game } from "./index";

test("My scenario 1", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.add("Clément");
    game.roll(1);
    game.wasCorrectlyAnswered(); // Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered(); // Thomas
    game.roll(1);
    game.wasCorrectlyAnswered(); // Clément
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
  });
});

test("My scenario 2", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(1);// Mathieu
    game.wasCorrectlyAnswered();
    game.roll(1);// Thomas
    game.wrongAnswer();
    game.roll(2);// Mathieu
    game.wasCorrectlyAnswered();
    game.roll(1);// Thomas
    game.wrongAnswer();
  });
});

test("My scenario 3", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(12);// Mathieu
    game.wrongAnswer();
    game.roll(3);// Thomas
    game.wrongAnswer();
    game.roll(2);// Mathieu
    game.wasCorrectlyAnswered();
    game.roll(9);// Thomas
    game.wrongAnswer();
    game.roll(3);// Mathieu
    game.wasCorrectlyAnswered();
  });
});

test("My scenario 4", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(1);
    game.wasCorrectlyAnswered(); // Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered(); // Thomas
    game.roll(1);
    game.wasCorrectlyAnswered();// Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();// Thomas
    game.roll(1);
    game.wasCorrectlyAnswered();// Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();// Thomas
    game.roll(1);
    game.wasCorrectlyAnswered();// Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();// Thomas
    game.roll(1);
    game.wasCorrectlyAnswered();// Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();// Thomas
    game.roll(1);
    game.wrongAnswer();         // Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();// Thomas
    game.roll(1);
    game.wasCorrectlyAnswered();// Mathieu
  });
});