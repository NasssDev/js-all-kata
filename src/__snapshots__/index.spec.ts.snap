// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My scenario 1 1`] = `
[
  "Player Mathieu has joined the game at place 0 with 0 point(s)",
  "Player Thomas has joined the game at place 0 with 0 point(s)",
  "Player Clément has joined the game at place 0 with 0 point(s)",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 1",
  "Science Question 0",
  "Player Mathieu + 1 point now at 1 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 1",
  "Science Question 1",
  "Player Thomas + 1 point now at 1 point(s)",
  "It is Clément's turn !",
  "player Clément has rolled 1 !",
  "player Clément is now at position 1",
  "Science Question 2",
  "Player Clément + 1 point now at 1 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 2",
  "Sports Question 0",
  "Player Mathieu + 1 point now at 2 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 2",
  "Sports Question 1",
  "Player Thomas + 1 point now at 2 point(s)",
  "It is Clément's turn !",
  "player Clément has rolled 1 !",
  "player Clément is now at position 2",
  "Sports Question 2",
  "Player Clément + 1 point now at 2 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 3",
  "Rock Question 0",
  "Player Mathieu + 1 point now at 3 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 3",
  "Rock Question 1",
  "Player Thomas + 1 point now at 3 point(s)",
  "It is Clément's turn !",
]
`;

exports[`My scenario 2 1`] = `
[
  "Player Mathieu has joined the game at place 0 with 0 point(s)",
  "Player Thomas has joined the game at place 0 with 0 point(s)",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 1",
  "Science Question 0",
  "Player Mathieu + 1 point now at 1 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 1",
  "Science Question 1",
  "Player Thomas has answered wrongly",
  "player Thomas goes in penalty box !",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 2 !",
  "player Mathieu is now at position 3",
  "Rock Question 0",
  "Player Mathieu + 1 point now at 2 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas goes out of penalty box ",
  "player Thomas is now at position 2",
  "Sports Question 0",
  "Player Thomas has answered wrongly",
  "player Thomas goes in penalty box !",
  "It is Mathieu's turn !",
]
`;

exports[`My scenario 3 1`] = `
[
  "Player Mathieu has joined the game at place 0 with 0 point(s)",
  "Player Thomas has joined the game at place 0 with 0 point(s)",
  "player Mathieu has rolled 12 !",
  "player Mathieu is now at position 0",
  "Pop Question 0",
  "Player Mathieu has answered wrongly",
  "player Mathieu goes in penalty box !",
  "It is Thomas's turn !",
  "player Thomas has rolled 3 !",
  "player Thomas is now at position 3",
  "Rock Question 0",
  "Player Thomas has answered wrongly",
  "player Thomas goes in penalty box !",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 2 !",
  "player Mathieu stays in penalty box",
  "It is Thomas's turn !",
  "player Thomas has rolled 9 !",
  "player Thomas goes out of penalty box ",
  "player Thomas is now at position 0",
  "Pop Question 1",
  "Player Thomas has answered wrongly",
  "player Thomas goes in penalty box !",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 3 !",
  "player Mathieu goes out of penalty box ",
  "player Mathieu is now at position 3",
  "Rock Question 1",
  "Player Mathieu has answered correctly",
  "Player Mathieu goes out of penalty box",
  "Player Mathieu + 1 point now at 1 point(s)",
  "It is Thomas's turn !",
]
`;

exports[`My scenario 4 1`] = `
[
  "Player Mathieu has joined the game at place 0 with 0 point(s)",
  "Player Thomas has joined the game at place 0 with 0 point(s)",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 1",
  "Science Question 0",
  "Player Mathieu + 1 point now at 1 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 1",
  "Science Question 1",
  "Player Thomas + 1 point now at 1 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 2",
  "Sports Question 0",
  "Player Mathieu + 1 point now at 2 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 2",
  "Sports Question 1",
  "Player Thomas + 1 point now at 2 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 3",
  "Rock Question 0",
  "Player Mathieu + 1 point now at 3 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 3",
  "Rock Question 1",
  "Player Thomas + 1 point now at 3 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 4",
  "Pop Question 0",
  "Player Mathieu + 1 point now at 4 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 4",
  "Pop Question 1",
  "Player Thomas + 1 point now at 4 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 5",
  "Science Question 2",
  "Player Mathieu + 1 point now at 5 point(s)",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 5",
  "Science Question 3",
  "Player Thomas + 1 point now at 5 point(s)",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu is now at position 6",
  "Sports Question 2",
  "Player Mathieu has answered wrongly",
  "player Mathieu goes in penalty box !",
  "It is Thomas's turn !",
  "player Thomas has rolled 1 !",
  "player Thomas is now at position 6",
  "Sports Question 3",
  "Player Thomas + 1 point now at 6 point(s)",
  "Player Thomas has won the Game !",
  "It is Mathieu's turn !",
  "player Mathieu has rolled 1 !",
  "player Mathieu goes out of penalty box ",
  "player Mathieu is now at position 7",
  "Rock Question 2",
  "Player Mathieu has answered correctly",
  "Player Mathieu goes out of penalty box",
  "Player Mathieu + 1 point now at 6 point(s)",
  "Player Mathieu has won the Game !",
  "It is Thomas's turn !",
]
`;
